### PR TITLE
Update documentation for Secrets Provider

### PR DESCRIPTION
**Pull request by rpothier**
### Desired Outcome

The README should include documentation for secrets provider

### Implemented Changes

Updated the README

### Connected Issue/Story

CyberArk internal issue link: [ONYX-20482](https://ca-il-jira.il.cyber-ark.com:8443/browse/ONYX-20482)

### Definition of Done
*At least 1 todo must be completed in the sections below for the PR to be
merged.*

#### Changelog

- [x] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [ ] This PR includes new unit and integration tests to go with the code
  changes, or
- [x] The changes in this PR do not require tests

#### Documentation

- [x] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]()
- [ ] This PR does not require updating any documentation

#### Behavior

- [ ] This PR changes product behavior and has been reviewed by a PO, or
- [ ] These changes are part of a larger initiative that will be reviewed later, or
- [x] No behavior was changed with this PR

#### Security

- [ ] Security architect has reviewed the changes in this PR,
- [ ] These changes are part of a larger initiative with a separate security review, or
- [x] There are no security aspects to these changes

### CODE REVIEW

**Review by szh — DISMISSED** [review]
LGTM!

File: README.md [hunk 1]
```diff
@@ -631,13 +631,14 @@ communicate with the Conjur appliance.
```
Comment (juniortaeza) [inline README.md:631]: Following our earlier conversation, a more fitting title prior to reaching this section might be something like:

"Conjur Authenticator / Secretless Sidecar / Secrets Provider Injection Examples"

or any shorter version :)

File: README.md [hunk 2]
```diff
@@ -631,13 +631,14 @@ communicate with the Conjur appliance.
...
-   1. [Deploying the Secretless Sidecar](#deploy-secretless-sidecar) 
+   2. [Deploying the Secretless Sidecar](#deploy-secretless-sidecar) or
+   3. [Deploying the Secrets Provider Sidecar](#deploy-secrets-provider-sidecar)
 
 ### Deploy Authenticator Sidecar
 
 1. Deploy an app with the Authenticator Sidecar:
     
-    The **Secretless Sidecar** is injected into the application pod on pod creation via the
+    The **Authenticator Sidecar** is injected into the application pod on pod creation via the
```
Comment (juniortaeza) [inline README.md:641]: good catch!

File: README.md [hunk 3]
```diff
@@ -817,6 +818,84 @@ be named secretless.
...
+   
+    ~$ cat << EOF | kubectl -n ${TEST_APP_NAMESPACE_NAME} apply -f -
+    apiVersion: v1
+    kind: Pod
+    metadata:
+      annotations:
+        conjur.org/inject: "true"
+        conjur.org/inject-type: "secrets-provider"
+        conjur.org/container-name: "cyberark-secrets-provider-for-k8s"
+        conjur.org/container-image: "docker.io//secrets-provider-for-k8s:edge"
```
Comment (juniortaeza) [inline README.md:844]: is it intended to have an empty string in this container-image path? i.e. in between docker.io and secrets-provider-fork8s:edge

"docker.io/ /secrets-provider-for-k8s:edge"

File: CHANGELOG.md [hunk 4]
```diff
@@ -17,6 +17,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
   [cyberark/sidecar-injector#72](https://github.com/cyberark/sidecar-injector/pull/72) <br>
   [cyberark/sidecar-injector#73](https://github.com/cyberark/sidecar-injector/pull/73) <br>
   [cyberark/sidecar-injector#74](https://github.com/cyberark/sidecar-injector/pull/74) <br>
+  [cyberark/sidecar-injector#79](https://github.com/cyberark/sidecar-injector/pull/79) <br>
 
 ### Changed
```
Comment (juniortaeza) [inline CHANGELOG.md:22]: should also add the details on what was changed/added

Comment (rpothier) [reply CHANGELOG.md:22]: the change goes with the above description, specific changes aren't needed for this.

File: README.md (hunk 3 above)
Comment (rpothier) [reply README.md:844]: no, thanks for finding it.

**Review by juniortaeza — APPROVED** [review]
LGTM!

**Pull request merged**